File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/views/login/EnterpriseCreateValidateView.java
package app.editors.manager.mvp.views.login;

import androidx.annotation.StringRes;

import moxy.viewstate.strategy.OneExecutionStateStrategy;
import moxy.viewstate.strategy.StateStrategyType;

import app.editors.manager.mvp.views.base.BaseView;

@StateStrategyType(OneExecutionStateStrategy.class)
public interface EnterpriseCreateValidateView extends BaseView {
    void onValidatePortalSuccess();
    void onPortalNameError(String message);
    void onEmailNameError(String message);
    void onFirstNameError(String message);
    void onLastNameError(String message);
    void onRegionDomain(String domain);
    void onShowWaitingDialog(@StringRes int title);
}

File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/views/login/EnterpriseCreateValidateView.kt
package app.editors.manager.mvp.views.login

import androidx.annotation.StringRes
import app.editors.manager.mvp.views.base.BaseView
import moxy.viewstate.strategy.OneExecutionStateStrategy
import moxy.viewstate.strategy.StateStrategyType

@StateStrategyType(OneExecutionStateStrategy::class)
interface EnterpriseCreateValidateView : BaseView {
    fun onValidatePortalSuccess(email: String?, first: String?, last: String?)
    fun onPortalNameError(message: String)
    fun onEmailNameError(message: String)
    fun onFirstNameError(message: String)
    fun onLastNameError(message: String)
    fun onRegionDomain(domain: String)
    fun onShowWaitingDialog(@StringRes title: Int)
}